src/lib.rs

```diff
@@ -273,6 +293,107 @@ impl OpenatDirExt for openat::Dir {
     }
 }
 
+fn map_nix_error(e: nix::Error) -> io::Error {
+    match e.as_errno() {
+        Some(os_err) => return io::Error::from_raw_os_error(os_err as i32),
+        _ => return io::Error::new(io::ErrorKind::Other, e),
+    }
+}
+
+fn copy_regfile_inner(
+    src: &File,
+    srcmeta: &openat::Metadata,
+    dest: &mut FileWriter,
+) -> io::Result<()> {
+    // Go directly to the raw file so we can reflink
+    let destf = dest.writer.get_mut();
+    let _ = src.copy_to(destf)?;
+    if srcmeta.stat().st_uid != nix::unistd::getuid().as_raw() {
```

Comment:
The `getuid()` here surprised me, and I suspect this condition fails to account for uid-vs-gid mismatches.
I would have expected something like `src.uid != dest.uid || src.gid != dest.gid`.